### ISSUE TITLE
North Star Onboarding

### ISSUE THREAD

**Issue by richtabor** [Automattic/wp-calypso#95367]
We would like to re-conceptualizing  a user's onboarding and in-product experience to fully reflect the breadth of what people can do. Instead of categorizing users by roles, we ask what they want to achieve, allowing for any combination of activities.  As described in p58i-irT-p2

This effort aims to provide a better understanding of what our customers are looking for, with next steps to correlate recommended themes, plugins, features, and launchpad tasks with these goals. 

We are starting with an update of the Goals screen and reconfiguration of the signup screens and will follow with an improved launchpad and suggestions of features, plugins, tools that a user may find valuable according to their Goals. 

![Image](https://github.com/user-attachments/assets/d35496e7-f324-438f-967b-5a3afecb56b1)

![Image](https://github.com/user-attachments/assets/3dbe48f1-9d19-4433-9b48-3a49022b750f)

----
Documentation and history of onboarding: PCYsg-D3w-p2

**Comment by simison** [comment]
> Launchpad: Tasks based on Goals selection.

Goals → Actions for writers: pfYzsZ-rr-p2

**Comment by p-jackson** [comment]
I found some great historical information about the difference between "intent" and "goals": pdDOJh-qB-p2

tl;dr: "Intent" is from a legacy UI where users chose an intent of either "writing", "building", or "selling". Users could only choose one intent. We now have "goals" which are stored as a serialised array so users can choose multiple. For backwards compatibility reasons we still synthesis an "intent" from the goals data so that any features that depended on "intent" continue to work.

[Here's the code that infers the intent given the selected goals](https://github.com/Automattic/wp-calypso/blob/02cd7d8d1d0b2210653f636137690efcf1d6f863/packages/data-stores/src/onboard/utils.ts#L12). It also appears that over time [the concept of "intent" has grown](https://github.com/Automattic/wp-calypso/blob/02cd7d8d1d0b2210653f636137690efcf1d6f863/packages/data-stores/src/onboard/constants.ts#L14-L33) to almost be synonymous with "flow".

For this task, let's add our new goals to [the `SiteGoals` enum](https://github.com/Automattic/wp-calypso/blob/02cd7d8d1d0b2210653f636137690efcf1d6f863/packages/data-stores/src/onboard/constants.ts#L3-L12), and for backwards compatibility have the new goals set an intent of `build`. For code that uses intent to determine the flow, I believe `build` will take us down the most generic onboarding flow.

Pinging for feedback from people who likely know more about goals/intents @Automattic/lego @Automattic/zap

**Comment by JanaMW27** [comment]
I don't seem to find an issue with the work related to moving the domains out of the onboarding. Will it be included here as a sub-issue? or should I create a new ticket?

**Comment by simison** [comment]
> For this task, let's add our new goals to [the `SiteGoals` enum](https://github.com/Automattic/wp-calypso/blob/02cd7d8d1d0b2210653f636137690efcf1d6f863/packages/data-stores/src/onboard/constants.ts#L3-L12), and for backwards compatibility have the new goals set an intent of `build`. For code that uses intent to determine the flow, I believe `build` will take us down the most generic onboarding flow.

What kind of backwards compatibility, specifically? What gets set downstream?

Since "build" goal is used for specific things, feels like least confusing path would be not to add it to every customer.

If there are mechanisms that should show up certain way (task lists, themes, plan grids) when any of the specific goals is picked, let's tie those into those goals then instead.

**Comment by autumnfjeld** [comment]
> Since "build" goal is used for specific things, feels like least confusing path would be not to add it to every customer.


Build is actually an `intent`.  Intent has evolved to essentially represent flows.  We'll need to keep the usage of `intent` to keep flows working as-is, perhaps deprecating the usage of `intent` slowly as we transition to using `site_intent`.

Code & documentation
* Field Guide: PCYsg-DE9-p2
* pfYzsZ-w2-p2
* https://github.com/Automattic/wp-calypso/blob/b423caff301109ae2de3d21c7bf2701d8c5a6e45/client/signup/steps/intent/types.ts
* https://github.com/Automattic/wp-calypso/blob/c27426f6ae9d170c567bf10b5825f97d211e1ea1/packages/data-stores/src/onboard/constants.ts#L3-L34

**Comment by wongasy** [comment]
Project Thread: pet6gk-1H4-p2

**Comment by chriskmnds** [comment]
> deprecating the usage of intent slowly as we transition to using site_intent.

I stumbled on some arbitrariness with `intent` and `site_intent` used in onboarding. It led me to create this issue: https://github.com/Automattic/wp-calypso/issues/93242

Not sure if there's any substance there, and I didn't have time to dig deeper.

**Comment by autumnfjeld** [comment]
@Lochlaer  Responding to the discussion in the where you went through this issue in p58i-jjR-p2.   You may have already gotten clarity, but as @richtabor touched on, the scope and complexity of this issue is big.      The North Star is a big vision laid out by Matias in p58i-irT-p2.

This issue encompasses a big vision with many projects (10-20 team project threads depending on how the work is broken down).   At least I have tried to break down the work in how it might be executed.    

I think what you/dotcom want to track on the dotcom board are the individual projects that have corresponding Project Threads. For example these active projects:
* [Focused Launchpad in My Home #98245](https://github.com/Automattic/wp-calypso/issues/98245)
* [Launchpad tasks: Create a Newsletter #98305](https://github.com/Automattic/wp-calypso/issues/98305)
* And soon Nexus projects will start
  * [Launchpad task: Create a course #98395](https://github.com/Automattic/wp-calypso/issues/98395) 
  * [Launchpad tasks: Upload videos #98396](https://github.com/Automattic/wp-calypso/issues/98396)

cc: @bohdankit07  @ouikhuan

**Comment by autumnfjeld** [comment]
@ianstewart  Can I remove this from  the dotcom projects board?  (Meaning remove Type Overview label).
 It isn't a project that can be tracked with ETA, it is a vision issue.

cc: @paulopmt1

**Comment by jboland88** [comment]
Moving this to icebox for the dotcom roadmap, but leaving it open for now. There may be pieces we'd like to reference going forward.
